### ISSUE TITLE
Updating Gradle and Lombok breaks Custom Task Annotations

### ISSUE THREAD

**Issue by physik932** [gradle/gradle#10997]
Hello!

I'm working on some old Java code that uses gradle and Lombok.  We're working on updating these dependencies to Gradle 5.6.2 and Lombok 1.18.10 and our custom task for integration tests is now failing.  Digging into the exceptions further, they're all based around various Lombok annotations.  

## What changed?
We use `gradlew`, so to update the Gradle version I updated it in the `gradle/wrapper/gradle-wrapper.properties` file:
```
distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.2-all.zip
distributionBase=GRADLE_USER_HOME
distributionPath=wrapper/dists
zipStorePath=wrapper/dists
zipStoreBase=GRADLE_USER_HOME
```

I changed our Lombok dependencies like so:

```
dependencies {
    ...
    // Lombok
    compileOnly("org.projectlombok:lombok:$lombokVersion")
    annotationProcessor("org.projectlombok:lombok:$lombokVersion")
    testCompileOnly("org.projectlombok:lombok:$lombokVersion")
    testAnnotationProcessor("org.projectlombok:lombok:$lombokVersion")
   ...
}
```

...where the `$lombokVersion` is `1.18.10`:
```
ext {
   ...
    lombokVersion = "1.18.10"
   ...
}
```

## What's broken?
However, this is when things started to go wrong with these tests.  Here's more background on the configuration and setup of the test.  

Here is the task:
```
task itest(type: Test) {
	testClassesDirs = sourceSets.itest.output.classesDirs
	classpath = sourceSets.itest.runtimeClassPath

	beforeTest { x -> logger.info("Running test: " + x) }

	// This is not needed, but I like to see which tests have run
	testLogging {
		events "passed", "skipped", "failed"
	}

	testlogger {
		// pick a theme - mocha, standard or plain
		theme 'standard'

		// set to false to disable detailed failure logs
		showExceptions true

		// set threshold in milliseconds to highlight slow tests
		slowThreshold 1000

		// displays a breakdown of passes, failures and skips along with total duration
		showSummary true

		// enable to see standard out and error streams inline with the test results
		showStandardStreams true
	}
}
```

Here is the `sourceSet` configuration:
```
sourceSets {
	itest {
		compileClasspath += main.output + configurations.testCompile
		runtimeClasspath += main.output + configurations.testRuntime
		java.srcDir file('src/itest/java')
	}
}
```

Here are the `configurations`:
```
configurations {
	itestCompile.extendsFrom testCompile
	itestCompileOnly.extendsFrom testCompileOnly
	itestRuntime.extendsFrom testRuntime
	itestAnnotationProcessor.extendsFrom testAnnotationProcessor
}
```

I believe what I'm looking for is something to configure the `sourceSets` using setAnnotationProcessorPath from [this document](https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/SourceSet.html#setAnnotationProcessorPath-org.gradle.api.file.FileCollection-).  Am I on the right track?  I can't get this running in my terminal or IntelliJ 2019.2.2.

Thank you all for your help.  If this is more appropriate for StackOverflow, I can post over there.  Thanks for your time.

**Comment by gregRzn** [comment]
@physik932 Hello,
Have you managed to overcome this problem? It seems that I'm facing the same issue and I'm out of ideas...

**Comment by physik932** [comment]
@gregRzn Sorry for this super late response.  I had another engineer look into this with me, and he found that we used a much newer version of Lombok at the time that seemed to change and break the way we used Gradle.  I believe it was something like 1.18.x; we downgraded to 1.16.x and it all worked again.  I know this is a late response and may not fix your issue.  Good luck and I hope you're doing well!
